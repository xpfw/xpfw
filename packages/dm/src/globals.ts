const StatType = {
  sum: "sum",
  mean: "mean",
  timeStep: "timeStep",
  sumTimeDistance: "sumTimeDistance",
  sumPrevTimeDistance: "sumPrevTimeDistance",
  avgTimeDistance: "avgTimeDistance",
  avgPrevTimeDistance: "avgPrevTimeDistance"
}

export {
  StatType
}
