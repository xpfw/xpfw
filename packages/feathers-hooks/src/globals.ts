const Method = {
  Create: "create",
  Update: "update",
  Patch: "patch",
  Find: "find",
  Remove: "remove",
  Get: "get"
}

export {
  Method
}
