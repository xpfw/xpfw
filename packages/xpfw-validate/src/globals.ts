const RequiredType = {
  Always: 2,
  AllNotSet: 3,
  SetToVal: 4
}

const ErrorType = {
  invalidType: 3,
  required: 4,
  tooLow: 5,
  tooHigh: 420,
  notEqual: 6,
  emptyUpdate: 7
}

const FieldType = {
  Text: 2,
  Number: 3,
  Boolean: 4,
  Date: 5,
  Select: 6,
  Password: 7,
  Array: 10,
  Object: 12,
  RelationshipSingle: 13,
  RelationshipMulti: 14,
  Slider: 15,
  Location: 16
}

const Permission = {
    Guest: "guest",
    User: "user",
    Owner: "owner",
    Admin: "admin",
    Server: "server"
}

const Method = {
  Create: "create",
  Update: "update",
  Patch: "patch",
  Find: "find",
  Remove: "remove",
  Get: "get"
}

const StatType = {
  sum: "sum",
  mean: "mean",
  timeStep: "timeStep",
  sumTimeDistance: "sumTimeDistance",
  sumPrevTimeDistance: "sumPrevTimeDistance",
  avgTimeDistance: "avgTimeDistance",
  avgPrevTimeDistance: "avgPrevTimeDistance"
}

const DateType = {
  dateTime: 2,
  date: 3,
  time: 4
}

const options = {
  userCollection: "users",
  userIdPath: "_id"
}

export {
  StatType, RequiredType, ErrorType, FieldType, Permission, Method, DateType, options
}
export default {
  StatType, RequiredType, ErrorType, FieldType, Permission, Method, DateType, options
}
