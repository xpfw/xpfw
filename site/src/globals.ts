import { StatType } from "@xpfw/dm"
import { executeForMethods, ExtendedJSONSchema } from "@xpfw/form"
import { isString } from "lodash"
import * as momentA from "moment"
import { changeValToRegex } from "./components/ui/regex"
const moment: any = momentA

const TagName: ExtendedJSONSchema = {
  title: "tagName",
  type: "string"
}
const TagDescription: ExtendedJSONSchema = {
  title: "description",
  type: "string"
}

const TagModel: ExtendedJSONSchema = {
  title: "tagModel",
  collection: "tags",
  required: [String(TagName.title)],
  properties: {
    [String(TagName.title)]: TagName,
    [String(TagDescription.title)]: TagDescription
  }
}

const Title: ExtendedJSONSchema = {
  title: "title",
  type: "string"
}

const CreatedAt: ExtendedJSONSchema = {
  title: "createdAt",
  type: "string",
  format: "date-time"
}

const Tags: ExtendedJSONSchema = {
  title: "tags",
  type: "array",
  theme: "multi",
  relationship: {
    namePath: TagName.title,
    collection: TagModel.collection,
    idPath: "_id"
  },
  modify: changeValToRegex(String(TagName.title))
}

const TagCollectionStats = [{
  id: "Some",
  type: StatType.sum,
  options: {itemPath: `${Tags.title}.length`}
}, {
  id: "Average",
  type: StatType.mean,
  options: {itemPath: `${Tags.title}.length`}
}, {
  id: "timeDistance",
  type: StatType.avgPrevTimeDistance,
  options: {itemPath: CreatedAt.title}
}, {
  id: "timeStepson",
  type: StatType.timeStep,
  options: {subType: StatType.sum, subConfig: {itemPath: `${Tags.title}.length`}}
}]

const ensureDate = (path: string, methods?: string[]) => {
  return executeForMethods((value, schema, method) => {
    if (isString(value[path])) {
      value[path] = moment(value[path]).toDate()
    }
    return Promise.resolve(value)
  }, methods)
}

const TagCollectionModel: ExtendedJSONSchema = {
  title: "tagColModel",
  collection: "tagCol",
  required: [String(Title.title)],
  properties: {
    [String(Title.title)]: Title,
    [String(Tags.title)]: Tags,
    [String(CreatedAt.title)]: CreatedAt
  },
  modify: [ensureDate(String(CreatedAt.title), ["create", "update"])]
}

const RecipeName: ExtendedJSONSchema = {
  title: "name",
  type: "string"
}

const RecipeAuthor: ExtendedJSONSchema = {
  title: "createdBy",
  type: "string"
}

const RecipeDate: ExtendedJSONSchema = {
  title: "createdAt",
  type: "object",
  format: "date-time"
}

const RecipeModel: ExtendedJSONSchema = {
  title: "recipeModel",
  collection: "recipes",
  required: [String(RecipeName.title), String(RecipeAuthor.title)],
  properties: {
    [String(RecipeName.title)]: RecipeName,
    [String(RecipeAuthor.title)]: RecipeAuthor
  }
}

const siteGlobals = {
  linkClass: "tag is-rounded is-white has-text-link is-size-6",
  contentClass: "is-size-5 smallMarginTop has-text-centered",
  externalLinkConfig: {
    className: "has-text-link",
    target: "_blank"
  },
  gitRoot: "https://github.com/xpfw/xpfw/",
  pkgRoot: "https://github.com/xpfw/xpfw/tree/master/packages/"
}

export default siteGlobals
export {
  TagModel, TagCollectionModel, TagName, TagDescription, Title, Tags,
  RecipeModel, RecipeName, RecipeAuthor, RecipeDate, CreatedAt, TagCollectionStats
}
