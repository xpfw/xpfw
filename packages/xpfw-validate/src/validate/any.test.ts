import * as MockDate from "mockdate"
import { ErrorType, FieldType } from "../globals"
import ValReg from "../registry"
import validateAnyE from "./any"

MockDate.set(new Date(2241, 3, 5))

const validateAny: any = validateAnyE
const customFieldType = "myCustomFieldForValidation"
ValReg.registerValidator(customFieldType, (value) => {
  return value === -1 ? Promise.resolve(value) :
    Promise.reject([{type: ErrorType.invalidType, mapTo: `constantlywrong`}])
})
test(`Full Coverage for validate/checkbox Form Validation`, (done) => {
  const validateWithCustom: any = {
    type: FieldType.Text,
    validate: {}
  }

  validateAny().then((validationRes: any) => { // Nothing to validate so true
    expect(validationRes).toMatchSnapshot("result of empty")
    return validateAny(null, {validate: {}, type: `invalid`})
  }).catch((e: any) => {
    expect(e).toMatchSnapshot(`validate unkown string type`)
    return validateAny(null, {validate: {}, type: false})
  }).catch((e: any) => {
    expect(e).toMatchSnapshot(`validate invalid boolean type`)
    return validateAny(null, {validate: {}, type: -1})
  }).catch((e: any) => {
    expect(e).toMatchSnapshot(`validate unkown number type`)
    validateWithCustom.validate.custom = () => {
      return Promise.resolve(`RESOLVEDVAL`)
    }
    return validateAny(32, validateWithCustom)
  }).catch((e: any) => {
    expect(e).toMatchSnapshot(`validate catched but custom ok`)
    return validateAny(`a`, validateWithCustom)
  }).then((e: any) => {
    expect(e).toMatchSnapshot(`custom and real validate ok`)
    validateWithCustom.validate.custom = () => {
      return Promise.reject([{text: `customreject`}])
    }
    return validateAny(`asdf`, validateWithCustom)
  }).catch((e: any) => {
    expect(e).toMatchSnapshot(`validate and custom not ok`)
    return validateAny(342,  {type: FieldType.Number, validate: {}})
  }).then((res: any) => {
    expect(res).toMatchSnapshot("Successfully validated any number")
    return validateAny(true,  {type: FieldType.Boolean, validate: {}})
  }).then((res: any) => {
    expect(res).toMatchSnapshot("Successfully validated any boolean")
    return validateAny(new Date(),  {type: FieldType.Date, validate: {}})
  }).then((res: any) => {
    expect(res).toMatchSnapshot("Successfully validated any date")
    return validateAny(["another", "string", "array"],  {type: FieldType.Array, validate: {type: FieldType.Text}})
  }).then((res: any) => {
    expect(res).toMatchSnapshot("Successfully validated any string array")
    return validateAny({my: true},  {type: FieldType.Object, validate: {
      objectDef: [{type: FieldType.Boolean, mapTo: "my"}]
    }})
  }).then((res: any) => {
    expect(res).toMatchSnapshot("Successfully validated any object")
    return validateAny(-1, {type: customFieldType, validate: {}})
  }).catch((res: any) => {
    expect(res).toMatchSnapshot("threw custom Field unkown error because no registry passed in params")
    return validateAny(-1, {type: customFieldType, validate: {}}, {registry: ValReg})
  }).then((res: any) => {
    expect(res).toMatchSnapshot("successfully validated custom Field")
    return validateAny(1, {type: customFieldType, validate: {}})
  }).catch((res: any) => {
    expect(res).toMatchSnapshot("successfully threw custom Field error")
    return validateAny(4,  {
      type: FieldType.Select, validate: {},
      selectOptions: [
        {label: "c", value: 4},
        {label: "a", value: 2}
      ]
    })
  }).then((res: any) => {
    expect(res).toMatchSnapshot("Successfully validated any select")
    return validateAny([1.3, 5.4],  {
      type: FieldType.Location, validate: {}
    })
  }).then((res: any) => {
    expect(res).toMatchSnapshot("Successfully validated any location")
    done()
  })
})
