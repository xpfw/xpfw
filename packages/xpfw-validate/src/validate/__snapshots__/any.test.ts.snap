// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Full Coverage for validate/checkbox Form Validation: Successfully validated any boolean 1`] = `true`;

exports[`Full Coverage for validate/checkbox Form Validation: Successfully validated any date 1`] = `2241-04-04T22:00:00.000Z`;

exports[`Full Coverage for validate/checkbox Form Validation: Successfully validated any location 1`] = `
Array [
  1.3,
  5.4,
]
`;

exports[`Full Coverage for validate/checkbox Form Validation: Successfully validated any number 1`] = `342`;

exports[`Full Coverage for validate/checkbox Form Validation: Successfully validated any object 1`] = `
Object {
  "my": true,
}
`;

exports[`Full Coverage for validate/checkbox Form Validation: Successfully validated any select 1`] = `4`;

exports[`Full Coverage for validate/checkbox Form Validation: Successfully validated any string array 1`] = `
Array [
  "another",
  "string",
  "array",
]
`;

exports[`Full Coverage for validate/checkbox Form Validation: custom and real validate ok 1`] = `"RESOLVEDVAL"`;

exports[`Full Coverage for validate/checkbox Form Validation: result of empty 1`] = `undefined`;

exports[`Full Coverage for validate/checkbox Form Validation: successfully threw custom Field error 1`] = `false`;

exports[`Full Coverage for validate/checkbox Form Validation: successfully validated custom Field 1`] = `-1`;

exports[`Full Coverage for validate/checkbox Form Validation: threw custom Field unkown error because no registry passed in params 1`] = `false`;

exports[`Full Coverage for validate/checkbox Form Validation: validate and custom not ok 1`] = `
Array [
  Object {
    "text": "customreject",
  },
]
`;

exports[`Full Coverage for validate/checkbox Form Validation: validate catched but custom ok 1`] = `
Object {
  "errors": Object {
    "errors": Array [
      Object {
        "type": 3,
      },
    ],
    "mapTo": undefined,
  },
  "mapTo": undefined,
}
`;

exports[`Full Coverage for validate/checkbox Form Validation: validate invalid boolean type 1`] = `false`;

exports[`Full Coverage for validate/checkbox Form Validation: validate unkown number type 1`] = `false`;

exports[`Full Coverage for validate/checkbox Form Validation: validate unkown string type 1`] = `false`;
